### DISCUSSION TITLE
Welcome to frappe_whatsapp Discussions!

### DISCUSSION THREAD

**Discussion by shridarpatil** [shridarpatil/frappe_whatsapp#6] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Atharva120818124** [comment]
I have install the app I have done all the meta developer setting, in the templets doctype Id and language code is also fetching properly but it is not working can you tell me where i am doing wrong or steps I am missing?

**Reply by shridarpatil** [reply]
Can you check the status of your template on meta? It must be approved.

**Reply by Atharva120818124** [reply]
yes
it is approve

**Reply by shridarpatil** [reply]
you can now setup notification. Also you are using dev credentials or prod?

**Reply by Atharva120818124** [reply]
actually it is Active Quality Pending

**Reply by shridarpatil** [reply]
its has to be approved

**Reply by Atharva120818124** [reply]
I have tried different templets but it is not giving approve it is giving Active Quality Pending can u tell me why?

**Reply by shridarpatil** [reply]
It takes around 24 hours to get approved from whats app. May I know when did you create template?

**Reply by Atharva120818124** [reply]
i have created yesterday it was in review state and today it is in Active Quality Pending

**Reply by shridarpatil** [reply]
Wait until its approved

**Reply by Atharva120818124** [reply]
can you tell me that the webhook configuration is mandatory or upto token generation is sufficent?

**Reply by shridarpatil** [reply]
No webhook configuration is not mandatory

**Comment by Atharva120818124** [comment]
![Screenshot from 2022-12-20 13-05-56](https://user-images.githubusercontent.com/110274134/208609846-aa253b20-241a-4c1d-9115-685d04623fb5.png)

**Comment by Atharva120818124** [comment]
i think active status should have done the job there is not approve status in the document i guess

**Reply by shridarpatil** [reply]
Ah yes your right. I got confused with the status.

**Reply by Atharva120818124** [reply]
can u plz give me some solution regarding my issue?

**Reply by shridarpatil** [reply]
Your using dev credentials or prod?

**Reply by Atharva120818124** [reply]
sorry I don't know what that means

**Reply by rknroot** [reply]
How to get production credentials in Meta.

**Reply by shridarpatil** [reply]
![image](https://user-images.githubusercontent.com/11792643/208831632-378926fb-8f95-4c2d-8309-1282c61b5f93.png)
change app mode from development to live

**Reply by rknroot** [reply]
I have enabled live mode, but how to get a permanent access token?
I got Temporary access token, that token will expire in 23 hours.

**Reply by Atharva120818124** [reply]
@rkr00t is your massage is sending from doctype

**Reply by rknroot** [reply]
@Atharva120818124 Nope, getting server error.
I think the problem is with the access token, If we get a permanent token problem will be solved.

**Reply by shridarpatil** [reply]
You will find it in your meta app configuration

**Comment by Atharva120818124** [comment]
![Screenshot from 2022-12-21 11-06-45](https://user-images.githubusercontent.com/110274134/208829672-981d34b0-ba91-4124-b0c2-e7e92c0ae3f7.png)
i don't know how but today i am getting this error

**Reply by shridarpatil** [reply]
can you show me your template?

**Comment by Atharva120818124** [comment]
![Screenshot from 2022-12-21 12-38-14](https://user-images.githubusercontent.com/110274134/208842721-5db1c229-cc84-4dcc-84a5-36a93e6d1dd3.png)

**Reply by shridarpatil** [reply]
And your notification?

**Comment by Atharva120818124** [comment]
![Screenshot from 2022-12-21 12-41-07](https://user-images.githubusercontent.com/110274134/208842988-83e1b06e-0d45-495f-a640-50ad0d9e50ac.png)

**Reply by Atharva120818124** [reply]
and i am using dev credentials

**Reply by shridarpatil** [reply]
For dev credential you need to register the number on meta for which your trying to send whats app message

**Reply by Atharva120818124** [reply]
can u guide me where to register the number

**Reply by Atharva120818124** [reply]
?

**Reply by Atharva120818124** [reply]
Now I have change from dev credential to prod

**Reply by shridarpatil** [reply]
After changing still your getting error?

**Reply by Atharva120818124** [reply]
actually error is gone but most important message is not sending id is generating properly

**Reply by shridarpatil** [reply]
I think your talking about template creation here. There is no ID generation during message sending.

**Reply by Atharva120818124** [reply]
exactly while template creation id is generating status of that template is Active but message is no sending

**Reply by shridarpatil** [reply]
After creating the template you need to set up a notification. And based on the event you select in notification message will get triggered

**Comment by buff0k** [comment]
Firstly, I want to thank you for the amazing app and almost seamless integration experience, will note that documentation is, perhaps, over-simplified but is not meant for someone without the background knowledge required.

I have a request for discussion on the point of setting up notifications, specifically, why the need to pull the notification number from the relevant doctype? Why can't we set specific telephone number for recipient (or, ideally, list of recipient numbers) when configuring the notification, this also brings me to the logical progression of allowing you to set up multiple recipients to a specific notification.

Example deployment - When a specific doctype is saved, the option in notification should not rely on any field in the doctype but rather send separate messages to all recipients listed.

**Reply by shridarpatil** [reply]
Sending notifications for every recipient is not ideal solution not everyone wants to send notification to every recipient.

Example: A sales invoice it dose'nt make sense to send everyone in the recipient list. And also it might increase the costing as the charges are message per person.

**Reply by buff0k** [reply]
Hence why I think a comma separated list of recipients would be beneficial, i.e. you can choose only one, or multiple recipients, this can be worked around if you can specify only one recipient per notification and create separate notifications for each recipient, but that would require you to be able to set the recipient number directly in the notification and not in the referenced DocType.

**Comment by resham29** [comment]
Hi 
I have a shared bench in frappe cloud and i need to start whatsapp intergration on my site how can i do it?

**Reply by shridarpatil** [reply]
@resham29  The shared bench doesn't support custom apps. You need to create a private bench and then install WhatsApp in your private bench.

**Reply by resham29** [reply]
Is there any alternate option for integrating whatsapp to frappe

**Comment by Meet-Tank4** [comment]
i have send whatsapp message not sends but failed how this solution 
<img width="964" height="810" alt="image" src="https://github.com/user-attachments/assets/f2cac0f9-04e9-44e0-8ddf-cdb10c4c6f37" />
